PublicMapping/districtbuilder
# Fix feature id conflict and avoid fitting map when splitting up a geounit

## Overview

Fix feature id conflict and avoid fitting map when splitting up a geounit

### Checklist


### Demo

![db_no_feature_id_conflict](https://user-images.githubusercontent.com/2926237/91456031-ca764e00-e850-11ea-92b0-591f4c2d05c0.gif)
![db_no_map_fit_bounds](https://user-images.githubusercontent.com/2926237/91456033-ca764e00-e850-11ea-8057-ec43a474cb82.gif)

### Notes
Includes moving frontend-only functions out of shared functions (see 4b6a55a) and upgrading prettier to support new operators used in the PR (namely optional chaining `?.`; see 5a0e4d1). FWIW I upgraded Prettier to the newest version (2.1+) and it reformatted tons of files so I decided to just upgrade to the latest minor release (looks like we'll have to add some configuration in the future to maintain our current formatting).

## Testing Instructions
* Go to a project using DE test data
* Select the first (northernmost) county
* Switch to tracts
* All tracts in the county should be selected _without fitting the map bounds_
* Assign all tracts to a given district
* Check that all tracts (including tract 0) were be assigned

Closes #307 
Closes #315
